dash_agent/README.md

```diff
@@ -1,39 +1,174 @@
+
+Sample example for `Command` object is shared below:
+
+```dart
+class AskCommand extends Command {
+  AskCommand({required this.docsSource});
+
+  final DataSource docsSource;
+
+  // Inputs
+  final userQuery = StringInput('Your query');
+  final codeAttachment = CodeInput('Code Attachment');
+
+ // Outputs
+ final matchingDocuments = MatchDocumentObject();
+ final queryOutput = QueryOutput();
+  
+  /// Unique identifier of the command
+  @override
+  String get slug => '/ask';
+
+  /// Brief description about the command
+  @override
+  String get intent => 'Ask me anything';
+
+  /// List of `DashInput`s that will be used in the command in its lifecycle
+  @override
+  List<DashInput> get registerInputs => [userQuery, codeAttachment];
+
+  /// Series of operation that needs to be performed for a command finish its task
+  @override
+  List<Step> get steps => [
+        MatchingDocumentStep(
+            query: '$userQuery$codeAttachment',
+            dataSources: [docsSource],
+            output: matchingDocuments),
+        PromptQueryStep(
+            prompt:
+                '''You are an X agent. Here is the $userQuery, here is the document references: $matchingDocuments. Answer the user's query.''',
+            postProcessKind: PostProcessKind.raw,
+            output: queryOutput),
+        AppendToChatStep(
+            value:
+                'This was your query: $userQuery and here is your output: $queryOutput'),
+      ];
+ 
+  /// Phrase that will be shown to user when the command is invoked
+  @override
+  String get textFieldLayout =>
+      "Hi, I'm here to help you. $userQuery $codeAttachment";
+}
+```
+
+One of important element of `Command` object are the series of steps that you will be passing that will help the command to execute its tasks by performing the mini-tasks required to be performed by the main task.
+
+
+### Step
+
+Currently supported steps that are avaiable for you leverage are shared below:
+
+- `MatchDocumentStep` - Helps you find the matching document from the provided data source form `DataSource` objects.
+- `WorkspaceQueryStep` - Helps you find the matching code snippets from the user's project.
+- `PromptQueryStep` - Enables you to perform a request to the lllm model with your customised prompt and instruction from the user to perform get generated code or any other general reponse that can be either used for next steps or passed back to the user as the final response.
+- `AppendToChatStep` - Enables you to append the reponse (anything like code, feedback, or general reponse) to the commanddash client chat box.
+- `ReplaceCodeStep` - This is experimental at the moment. This step will enable you to directly replace the user's selected code with the resulting output code from the command.
```

Comment:
```suggestion
```

Reply:
@samyakkkk Can you please help me understand the reason behind recommending this update!

Reply:
@zexross yes, it's not ready to be used by users (even experimentally) at this point. So best to not mention.